xonsh/built_ins.py

```diff
@@ -555,6 +564,16 @@ def run_subproc(cmds, captured=True):
                 except PermissionError:
                     e = 'xonsh: subprocess mode: permission denied: {0}'
                     raise XonshError(e.format(cmd[0]))
+        _stdin_file = None
+        if captured == 'object' and stdin is not None:
+            _stdin_file = tempfile.NamedTemporaryFile()
+            cproc = Popen(['cat'],
+                          stdin=stdin,
+                          stdout=PIPE)
+            tproc = Popen(['tee', _stdin_file.name],
```

Comment:
I don't think that we capture stdin elsewhere, do we?  I tend to think of capturing stdin as a potential security whole as it could easily capture things like passwords and private keys.  If you want to do a pure python version, that would be fine, but there should probably be an envvar that toggles this behaviour (`$XONSH_STORE_STDIN` perhaps) that is off by default.